lib/openhab/log/logger.rb

```diff
@@ -79,7 +79,7 @@ def clean_backtrace(error)
           backtrace = error.stack_trace.reject { |line| JAVA_INTERNAL_CALL_REGEX.match? line.to_s }
           error.set_stack_trace(backtrace)
         end
-        error
+        error.is_a?(Exception) ? error : JavaException.new(error.message)
```

Comment:
I'd recommend passing the original error to JavaException.new, and then in an initialize method in that class store the entire error to an ivar, and only pass the message to super. You should also look if jruby already has a similar Ruby class defined

Reply:
Thanks for your suggestion. What should be done with that ivar?

The first problem is that I don't know how to reproduce this issue. So far I very rarely (but not never) encountered it on my production system. However, I see this error most often in the cucumber tests on the github workflow.

Also upon further investigation, this `error` came from `StandardError => e`

https://github.com/boc-tothefuture/openhab-jruby/blob/409c7a07586dc0a2cdf828317a72790a90ccf51a/lib/openhab/dsl/rules/rule.rb#L51

So it seems that it should've already been an instance of StandardError, but why re-raising it can cause 
```
(TypeError) exception class/object expected
```

Reply:
Just accessible with an attr_reader. As for how it could happen... no idea.

Reply:
Maybe add a rescue around the raise that if it does a TypeError, log the details of what you're trying to raise and then re-raise (the TypeError).

Perhaps there's a funny jruby thing going on that StandardError#=== is catching something Java-ish, but internally it can't re-raise

Reply:
Who is going to consume / use that attribute? Do I just pick an arbitrary name, e.g. `exception_object` or something?

Reply:
Anyone interested in the original error. On second thought, there might be an implicit `cause` accessor that's populated. Please to double check.

Reply:
> Maybe add a rescue around the raise that if it does a TypeError, log the details of what you're trying to raise and then re-raise (the TypeError).

That's actually a brilliant idea!

Reply:
I've removed the JavaException class for now and just log the error that caused the TypeError as you suggested.